src/connectors/nginx.py

```diff
@@ -0,0 +1,290 @@
+        'name': 'existing_stage',
+        'title': "Snowflake Stage (alternative)",
+        'prompt': "Enter to use an existing stage instead",
+        'placeholder': "snowalert.data.nginx_stage",
+    },
+]
+
+
+LOG_LANDING_TABLE_COLUMNS = [
+    ("raw", "VARIANT"),
+    ("hash_raw", "NUMBER(19,0)"),
+    ("request_id", "VARCHAR(16777216)"),
+    ("event_time", "TIMESTAMP_LTZ(9)"),
+    ("bytes_sent", "NUMBER(38,0)"),
+    ("connection", "NUMBER(38,0)"),
+    ("connection_requests", "NUMBER(38,0)"),
+    ("deployment_cluster", "VARCHAR(16777216)"),
+    ("gzip_ratio", "FLOAT"),
+    ("host_header", "VARCHAR(16777216)"),
+    ("host_name", "VARCHAR(16777216)"),
+    ("user_agent", "VARCHAR(16777216)"),
+    ("http_user_agent", "VARCHAR(16777216)"),
+    ("http_xff", "VARCHAR(16777216)"),
+    ("http_referer", "VARCHAR(16777216)"),
+    ("http_method", "VARCHAR(16777216)"),
+    ("instance_id", "VARCHAR(16777216)"),
+    ("redirect_counter", "NUMBER(38,0)"),
+    ("remote_address", "VARCHAR(16777216)"),
+    ("request", "VARCHAR(16777216)"),
+    ("request_time", "FLOAT"),
+    ("requests_length", "NUMBER(38,0)"),
+    ("ssl_session_id", "VARCHAR(16777216)"),
+    ("ssl_session_reused", "VARCHAR(16777216)"),
+    ("status", "NUMBER(38,0)"),
+    ("event_time_other", "VARIANT"),
+    ("n_upstream_attempts", "NUMBER(38,0)"),
+    ("upstream_address", "VARCHAR(16777216)"),
+    ("upstream_response_length", "NUMBER(38,0)"),
+    ("upstream_response_time", "FLOAT"),
+    ("upstream_status", "NUMBER(38,0)"),
+]
+
+ERROR_LANDING_TABLE_COLUMNS = [
+    ("raw", "VARIANT"),
+    ("hash_raw", "NUMBER(19,0)"),
+    ("instance_id", "VARCHAR(16777216)"),
+    ("log_level", "VARCHAR(16777216)"),
+    ("message", "VARCHAR(16777216)"),
+    ("event_time", "TIMESTAMP_LTZ(9)"),
+    ("pid", "NUMBER(38,0)"),
+    ("tid", "NUMBER(38,0)"),
+]
+
+CONNECT_RESPONSE_MESSAGE = """
+STEP 1: Be sure the Role "{role}" includes the following trust relationship:
+
+{role_trust_relationship}
+
+
+STEP 2: For Role "{role}", ensure the following inline policy:
+
+{role_policy}
+"""
+
+
+def connect(connection_name, options):
+    table_name = f'nginx_{connection_name}_connection'
+    log_landing_table = f'data.{table_name}'
+    error_landing_table = f'data.nginx_{connection_name}_error_connection'
+    prefix = ''
+    bucket_name = ''
+
+    for i in options:
+        if i == '':
+            options.pop(i)
+
+    db.create_table(
+        name=log_landing_table,
+        cols=LOG_LANDING_TABLE_COLUMNS,
+        comment=yaml_dump(module='nginx', **options)
+    )
+    db.execute(f'GRANT INSERT, SELECT ON {log_landing_table} TO ROLE {SA_ROLE}')
+
+    db.create_table(
+        name=error_landing_table,
+        cols=ERROR_LANDING_TABLE_COLUMNS,
+        comment=yaml_dump(module='nginx', **options)
+    )
+    db.execute(f'GRANT INSERT, SELECT ON {error_landing_table} TO ROLE {SA_ROLE}')
+
+    stage = options.get('existing_stage')
+    if stage:
+        stage_name = stage
```

Comment:
should this also clear out the other options as it does in the ldap connector?

Reply:
Yes. I wrote this one before I wrote that one, but I should bring that functionality over here as well. It means that if you provide an existing stage, you don't have to clear out the defaults from fields which provide one; it'll get automatically discarded (because we can derive everything we need from the existing stage and it's better to do that than ask users to remember what the stage says and risk them giving us the wrong information).

Reply:
Looking at the code, we actually already do this; those values are initialized to empty strings at lines 108 and 109, and only populated with the options if we don't get an existing stage. And then the only time we use the AWS_ROLE option is when we're creating a new stage; the rest of the time, we get the role from the stage itself. So I think this one is actually fine as-is.

Reply:
sounds good. to make it a little more standard / clear, could we use the idiom I just changed `osquery_logs` to, i.e.

    stage_name = options.get('existing_stage')
    if not stage_name:
        ...